### PR TITLE
feat(job): add job waiting option

### LINKED ISSUE #723: feat: blocking job wait for scripting (lily wait job --id N)
### Description

I was trying to automate the successive execution of multiple jobs. I ended up with a brittle bash/jq script which depends on the job list output not changing. Would be really useful if we could script something like `lily job wait --id N` which would block until the job completed. Even better if it spits out the job status once it's done so things like `Error` could be inspected.

### PR DESCRIPTION
**Pull request by frrist**
- closes #723
- adds a new test `Job Submit and wait`
- refactors the scheduler test, can add as a separate change if prefers.

### CODE REVIEW

File: schedule/scheduler.go [hunk 1]
```diff
@@ -290,6 +291,40 @@ func (s *Scheduler) StopJob(id JobID) error {
 	return nil
 }
 
+func (s *Scheduler) WaitJob(id JobID) (*JobListResult, error) {
+	s.jobsMu.Lock()
+	job, ok := s.jobs[id]
+	if !ok {
+		return nil, xerrors.Errorf("wait job ID: %d not found", id)
```
Comment (iand) [inline schedule/scheduler.go:298]: Unlock `jobsMu` before return. Can just move the existing `Unlock` to just before the `if`

File: schedule/scheduler.go [hunk 2]
```diff
@@ -290,6 +291,40 @@ func (s *Scheduler) StopJob(id JobID) error {
...
+	// wait on the job to complete
+	<-job.Job.Done()
+
+	// refetch the job to get latest results (EndedAt and Running will have changed)
+	s.jobsMu.Lock()
+	job, ok = s.jobs[id]
+	if !ok {
+		return nil, xerrors.Errorf("wait job ID: %d not found", id)
+	}
+	s.jobsMu.Unlock()
```
Comment (iand) [inline schedule/scheduler.go:311]: Same here, make sure jobsMu is unlocked before return

**Review by iand — CHANGES_REQUESTED** [review]

File: schedule/scheduler.go (hunk 1 above)
Comment (frrist) [reply schedule/scheduler.go]: whoops, done.

File: schedule/scheduler.go (hunk 2 above)
Comment (frrist) [reply schedule/scheduler.go]: done.

**Review by iand — APPROVED** [review]

**Pull request merged**